File: backend/app/models/email.py
from pydantic import BaseModel, Field
from enum import Enum


class RecipientContext(str, Enum):
    colleague = "colleague"
    manager = "direct manager"
    executive = "senior executive"
    client = "external client"
    new_contact = "new business contact"


class EmailRequest(BaseModel):
    message: str = Field(..., min_length=2, max_length=1000)
    recipient_context: RecipientContext = RecipientContext.colleague


class EmailVariant(BaseModel):
    subject: str
    body: str
    explanation: str
    politeness_level: int = Field(..., ge=1, le=5)


class EmailResponse(BaseModel):
    formal: EmailVariant
    keigo: EmailVariant
    sonkeigo: EmailVariant
    cultural_tip: str

File: frontend/src/types/email.ts
export type RecipientContext =
  | "colleague"
  | "direct manager"
  | "senior executive"
  | "external client"
  | "new business contact";

export interface EmailRequest {
  message: string;
  recipient_context: RecipientContext;
}

export interface EmailVariant {
  subject: string;
  body: string;
  explanation: string;
  politeness_level: number;
}

export interface EmailResponse {
  formal: EmailVariant;
  keigo: EmailVariant;
  sonkeigo: EmailVariant;
  cultural_tip: string;
}

export type ToneKey = "formal" | "keigo" | "sonkeigo";

export interface ToneOption {
  key: ToneKey;
  label: string;
  jp: string;
  desc: string;
}
